src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -25,7 +26,6 @@ public class DefaultBeanFactory implements BeanFactory {
 
     public DefaultBeanFactory(Map<String, BeanDefinition> beanDefinitions) {
         this.beanDefinitions = beanDefinitions;
-        initializeBeans();
     }
```

Comment:
Perhaps, we can add a method like `registerBeanDefinition` or `addBeanDefinition` and fill the `beanDefinitions` up by passing constructed `BeanDefinition`s one by one from the caller side instead of passing the Map into the constructor.

```
void regregisterBeanDefinition(String name, BeanDefinition beanDefinition) {...}
```
What do you think?